dask_jobqueue/sge.py

```diff
@@ -62,8 +65,16 @@ def __init__(self, queue=None, project=None, resource_spec=None, walltime=None,
             header_lines.append('#$ -q %(queue)s')
         if project is not None:
             header_lines.append('#$ -P %(project)s')
+
+        # Define resource specifications. There are only two cases allowed in this
+        # implementation: either specify using resource_spec, or specify using the rest of the
+        # kwargs.
         if resource_spec is not None:
             header_lines.append('#$ -l %(resource_spec)s')
+        else:
+            header_lines.append('#$ -l m_mem_free=%s' % parse_memory(kwargs.get('memory')))
```

Comment:
I'm wondering if you could add CPU resources spec too, even if this is not the original problem, shouldn't be too hard or introduce much noise.

Reply:
I'd like to hold off on that, but only for this PR. Happy to do it in a separate PR, so that we can settle issues independently.

Reply:
Ok, fine by me.

Reply:
`m_mem_free` does not exist on my SGE cluster. Quickly googling this may only exist for Univa Grid Engine.
On my cluster there is `mem_req` that could be used. Does that exist on yours @ericmjl?

Reply:
It does not look like `mem_req` is standard at all ... not sure what to do.

Reply:
@lesteve I am starting to discover this as well. It appears that mapping the memory keyword argument to any particular configuration string is not going to be productive.

I am beginning to think I should close this PR and open a different PR for documentation purposes.

Reply:
Wow, that's good to know! It there is no memory option standard, then yes, no other choice to keep with `resources_spec` and document this.